### ISSUE TITLE
Ability to change tab indentation size

### ISSUE THREAD

**Issue by DamianSuess** [aelij/RoslynPad#358]
**Describe the solution you'd like**
As a user, I would like to be able to change the indentation spaces to match our personal settings. For example, 2 spaces instead of 4 when pressing the tab key.

**Comment by DamianSuess** [comment]
@aelij, do you have a preference for a strategy on environment configurations?

As per the PR by @mitchcapper and the comment made in the enhancement request [330](https://github.com/roslynpad/roslynpad/issues/330#issuecomment-586723111), I'm curious as to your thoughts on the direction of a Preferences configuration, even if its strictly aimed at AvalonEdit or perhaps the use of a basic `.editorconfig` file in the same folder as the `.csx` files.

Side note, I wouldn't mind making a request for a dark theme setting.. the button just needs a home (even if it's the main screen).

**Comment by aelij** [comment]
EditorConfig is supported in v17

**Comment by DamianSuess** [comment]
@aelij wait a minute.. it was as easy as just passing it into `RoslynHost` to send to the `Solution`s  object` `ProjectInfo`?

Wow, that's amazing. I didn't know Roslyn's CodeAnalysis had that feature built into it. Previously, I was looking into translating the `.ini` and sending it over. Sir, that is an elegant solution 👍 

```cs
RoslynHost = await Task.Run(() => new RoslynHost(CompositionAssemblies,
            RoslynHostReferences.NamespaceDefault.With(imports: new[] { "RoslynPad.Runtime" }),
            disabledDiagnostics: ImmutableArray.Create("CS1701", "CS1702", "CS7011", "CS8097"),
            analyzerConfigFiles: ImmutableArray.Create(_editorConfigPath)))
            .ConfigureAwait(true);
```

**Comment by aelij** [comment]
Yes, it has to be supported at the compiler level because indentation depends on the language's syntax.

Also it needed to be hooked up to the editor:
https://github.com/roslynpad/roslynpad/blob/73372a15821287161dbaaf4415b81daef44f17eb/src/RoslynPad.Editor.Shared/RoslynCodeEditor.cs#L105-L110

**Comment by DamianSuess** [comment]
ahhh, there's the magic glue! Thank you for sharing
